libsolidity/interface/CompilerStack.cpp

```diff
@@ -225,11 +226,12 @@ bool CompilerStack::parse()
 	if (m_stackState != SourcesSet)
 		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Must call parse only after the SourcesSet state."));
 	m_errorReporter.clear();
-	ASTNode::resetID();
 
 	if (SemVerVersion{string(VersionString)}.isPrerelease())
 		m_errorReporter.warning("This is a pre-release compiler version, please do not use it in production.");
 
+	Parser parser{m_errorReporter, m_evmVersion, m_parserErrorRecovery};
```

Comment:
This seems to be an unrelated change to this PR. Why was this done?

Reply:
We moved the NodeID counter into `Parser`. We don't want it to reset every time we processed a file, so we moved it outside the loop to keep the counter state.

Reply:
Is moving the node id counter still necessary?

Reply:
I think so, why wouldn't it be? Before this PR the counter is singleton - that's always in contrast to having it in the parser and I don't see any reason for not moving it there. And it still seems the cleanest solution for allowing both parsing ids and importing ids.

Reply:
ok!